File: app/src/main/java/org/tasks/injection/InjectingBottomSheetDialogFragment.java
package org.tasks.injection;

import android.app.Activity;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

public abstract class InjectingBottomSheetDialogFragment extends BottomSheetDialogFragment {

  private boolean injected;

  @Override
  public void onAttach(Activity activity) {
    super.onAttach(activity);

    if (!injected) {
      inject(((InjectingActivity) activity).getComponent().plus(new DialogFragmentModule(this)));
      injected = true;
    }
  }

  protected abstract void inject(DialogFragmentComponent component);
}

File: app/src/main/java/org/tasks/injection/InjectingBottomSheetDialogFragment.kt
package org.tasks.injection

import android.app.Activity
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

abstract class InjectingBottomSheetDialogFragment : BottomSheetDialogFragment() {
    private var injected = false
    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        if (!injected) {
            inject((activity as InjectingActivity).component.plus(DialogFragmentModule(this)))
            injected = true
        }
    }

    protected abstract fun inject(component: DialogFragmentComponent)
}